cogs/welcome/welcome.py

```diff
@@ -132,6 +198,144 @@ async def logServerLeave(self, leaveUser: discord.Member):
     async def welcome(self, ctx: Context):
         """Server welcome message settings."""
 
+    @checks.mod_or_permissions()
+    @welcome.command(name="channel")
+    async def welcomeChannelSet(self, ctx: Context, channel: discord.TextChannel = None):
+        """
+        Set the welcome channel
+
+        Parameters:
+        -----------
+        channel: The text channel to set welcome's to. If not passed anything, will remove the welcome channel
+        """
+        if not channel:
+            # channel == None
+            await self.config.guild(ctx.guild).welcomeChannelSet.set(False)
+            await ctx.send("Welcome channel has been removed")
+            return
+
+        await self.config.guild(ctx.guild).welcomeChannel.set(channel.id)
+        await self.config.guild(ctx.guild).welcomeChannelSet.set(True)
+        await ctx.send(f"Channel set to {channel}")
+
+        return
+
+    @checks.mod_or_permissions()
+    @welcome.command(name="add")
+    async def greetAdd(self, ctx: Context, name: str):
+        """
+        Add a new greeting
+
+        Including {USER} in the message will have that replaced with a ping to the new user
+
+        Parameters:
+        -----------
+        name: name of the greeting
+        """
+
+        def check(message: discord.Message):
+            return message.author == ctx.message.author and message.channel == ctx.message.channel
+
+        await ctx.send("What would you like the greeting message to be?")
+        try:
+            greeting = await self.bot.wait_for("message", check=check, timeout=30.0)
+        except asyncio.TimeoutError:
+            await ctx.send("No response received, not setting anything!")
+            return
+
+        if len(greeting.content) > 2048:
+            await ctx.send("Your message is too long!")
+            return
```

Comment:
Opps, I think the place we copied from was actually incorrect before: max message length is 2000 (can make 2000 a constant). Also, I think it might be good to make the length check with a temporary variable where we replace `{USER}` with the at-mention. The at-mention replacement will be longer than `{USER}` because it's replaced with something like `<@0000000000000>`, so we still might go over.

Reply:
done